### PR TITLE
replaced search cache

### PR DESCRIPTION
**Pull request by Ark20**
errors are not cached #1235

### CODE REVIEW

File: src/backend/tests/factories/crawler_factory.py [hunk 1]
```diff
@@ -38,5 +38,5 @@ def create_ambiguous_record_with_questions(
 
             aliases = (Alias(first_name="John", last_name="Doe", middle_name="", birth_date=""),)
             return RecordCreator.build_record(
-                RecordCreator.build_search_results.__wrapped__, "username", "password", aliases, {}
```
Comment (wittejm) [inline src/backend/tests/factories/crawler_factory.py:41]: I think this will do it?

File: src/backend/tests/factories/crawler_factory.py [hunk 2]
```diff
@@ -38,5 +38,5 @@ def create_ambiguous_record_with_questions(
 
             aliases = (Alias(first_name="John", last_name="Doe", middle_name="", birth_date=""),)
             return RecordCreator.build_record(
```
Comment (wittejm) [inline src/backend/tests/factories/crawler_factory.py:40]:
```suggestion
            record = RecordCreator.build_record(
                RecordCreator.build_search_results, "username", "password", aliases, {}
            )
            RecordCreator.search_cache = LRUCache(4) # clear the cache for each test case
            return record
```

Comment (wittejm) [reply src/backend/tests/factories/crawler_factory.py:40]: well. Actually that will probably break the test case that tests that the cache works.... what a finicky issue.

File: src/backend/expungeservice/record_creator.py [hunk 3]
```diff
@@ -59,23 +62,29 @@ def build_search_results(
...
-            except InvalidOECIUsernamePassword as e:
-                error(401, str(e))
-            except OECIUnavailable as e:
-                error(404, str(e))
-            except Exception as e:
-                errors.append(str(e))
-            finally:
-                session.close()
-        return search_results, errors
+        alias_match = RecordCreator.search_cache.__getitem__(aliases)
```
Comment (KentShikama) [inline src/backend/expungeservice/record_creator.py:65]: You can use the syntax `RecordCreator.search_cache[aliases]`. See [`__getitem__`](https://docs.python.org/3/reference/datamodel.html#object.__getitem__).

File: src/backend/tests/endpoints/test_search.py [hunk 4]
```diff
@@ -109,3 +116,32 @@ def test_search_with_failing_save_event(service, monkeypatch):
...
+    successful search results should cache
+    """
+    monkeypatch.setattr(Crawler, "attempt_login", mock_login("Successful login response"))
+    monkeypatch.setattr(Crawler, "search", mock_search(service, "john_doe"))
+
+    test_alias=(Alias("john","deer", "",""),)
+
+    RecordCreator.build_search_results("username", "password", test_alias )
+
+    assert RecordCreator.search_cache.__getitem__(test_alias)
```
Comment (KentShikama) [inline src/backend/tests/endpoints/test_search.py:132]: Similar to above.

File: src/backend/tests/endpoints/test_search.py [hunk 5]
```diff
@@ -109,3 +116,32 @@ def test_search_with_failing_save_event(service, monkeypatch):
...
+    """
+
+    monkeypatch.setattr(Crawler, "attempt_login", mock_login("Successful login response"))
+    monkeypatch.setattr(Crawler, "search", mock_search_fail())
+
+    test_fail_alias = (Alias("jake","doe", "",""),)
+
+    RecordCreator.build_search_results("username","password", test_fail_alias)
+
+    assert not RecordCreator.search_cache.__getitem__(test_fail_alias)
```
Comment (KentShikama) [inline src/backend/tests/endpoints/test_search.py:147]: Run `pre-commit install`. You may need to `pip install pre-commit`.

Comment (KentShikama) [reply src/backend/tests/endpoints/test_search.py:147]: Also similar to above with `__getitem__`.

File: src/backend/tests/endpoints/test_search.py [hunk 6]
```diff
@@ -109,3 +116,32 @@ def test_search_with_failing_save_event(service, monkeypatch):
     data = json.loads(response.data)
 
     check_response_record_matches_mock_crawler_search(data["record"], service.mock_record["john_doe"])
+
+
+def test_search_cache(service, monkeypatch): 
+    """
+    successful search results should cache
```
Comment (KentShikama) [inline src/backend/tests/endpoints/test_search.py:123]: This comment is unnecessary. Perhaps just rename the test `test_successful_search_cache`.

File: src/backend/tests/endpoints/test_search.py [hunk 7]
```diff
@@ -109,3 +116,32 @@ def test_search_with_failing_save_event(service, monkeypatch):
...
+    test_alias=(Alias("john","deer", "",""),)
+
+    RecordCreator.build_search_results("username", "password", test_alias )
+
+    assert RecordCreator.search_cache.__getitem__(test_alias)
+
+
+def test_search_cache_error(service, monkeypatch): 
+    """
+    Search shouldn't cache when there is an error
```
Comment (KentShikama) [inline src/backend/tests/endpoints/test_search.py:137]: This comment is unnecessary.

**Pull request merged**